examples/example-01.device.nut

```diff
@@ -1,77 +1,89 @@
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+// "Promise" symbol is injected dependency from ImpUnit_Promise module,
+// while class being tested can be accessed from global scope as "::Promise".
+
+#require "MessageManager.lib.nut:2.0.0"
+#require "SPIFlashFileSystem.class.nut:2.0.0"
+
+// Initialize using default settings
+local mm = MessageManager();
+spiffs <- SPIFlashFileSystem();
+spiffs.init();
 
-// Function to recursively get the next chunk of the URL until it is finished
 function getNextChunk(file = null, offset = 0, length = 1000) {
 
     // Prepare a request packet
     local request = {
-        "url": "https://pbs.twimg.com/profile_images/435528858532470784/alH9RtCl_400x400.jpeg",
+        "url": "https://electricimp.com/public/v3/apple-touch-icon.png",
         "offset": offset,
         "length": length
     }
 
     // Send the request to the agent
-    bull.send("http.get", request)
-
-        .onReply(
-
-            // We received a response
-            function(message) {
-
-                local done = false;
-
-                if (message.data) {
-                    // We have a chunk of data
-
-                    // Append it to the file
-                    if (!file) file = spiffs.open("electricimp.jpg", "w");
-                    file.write(message.data)
-
-                    if (message.data.len() == length) {
-                        // Request the next chunk
-                        getNextChunk(file, offset + length, length);
-                    } else {
-                        // We have more or less than we asked for. Best to stop
-                        done = true;
-                    }
-
-                } else if (file) {
-                    // We have finished or there is an error
-                    done = true;
-                }
-
-                if (done) {
-                    // Close the file
-                    server.log("Received file of length: " + file.len())
-                    file.close();
-                    file = null;
-                }
-
-            }.bindenv(this)
-
-        )
-
-        .onFail(
-
-            // The request timed out or failed, just log and retry
-            function(err, message, retry) {
-                server.error("Failed get next chunk. Retrying.")
-                retry(10);
-            }.bindenv(this)
-
-        )
-
+    mm.send("http.get", request,
+      {"onReply": function(msg, response) {
```

Comment:
There is something wrong with indentation here. It should be 4 spaces.

And also please use this format:

```
mm.send("http.get", request,
    {  <------ goes on a separate line
        "onReply": function(msg, response) {
        }.bindenv(this),
        ...
    }
```